src/Motor.Extensions.Hosting/MultiOutputServiceAdapter.cs

```diff
@@ -16,35 +13,29 @@ public class MultiOutputServiceAdapter<TInput, TOutput> : INoOutputService<TInpu
 {
     private readonly IMultiOutputService<TInput, TOutput> _converter;
     private readonly ILogger<SingleOutputServiceAdapter<TInput, TOutput>> _logger;
-    private readonly ISummary? _messageProcessing;
     private readonly ITypedMessagePublisher<TOutput> _publisher;
 
     public MultiOutputServiceAdapter(ILogger<SingleOutputServiceAdapter<TInput, TOutput>> logger,
-        IMetricsFactory<SingleOutputServiceAdapter<TInput, TOutput>>? metrics,
         IMultiOutputService<TInput, TOutput> converter,
         ITypedMessagePublisher<TOutput> publisher)
     {
         _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         _converter = converter ?? throw new ArgumentNullException(nameof(converter));
         _publisher = publisher ?? throw new ArgumentNullException(nameof(publisher));
-        _messageProcessing = metrics?.CreateSummary("message_processing", "Message processing duration in ms");
     }
 
     public async Task<ProcessedMessageStatus> HandleMessageAsync(MotorCloudEvent<TInput> dataCloudEvent,
         CancellationToken token = default)
     {
         try
         {
-            using (new AutoObserveStopwatch(() => _messageProcessing))
+            await foreach (var message in _converter.ConvertMessageAsync(dataCloudEvent, token)
+                               .ConfigureAwait(false).WithCancellation(token))
             {
-                await foreach (var message in _converter.ConvertMessageAsync(dataCloudEvent, token)
-                                   .ConfigureAwait(false).WithCancellation(token))
+                if (message?.Data is not null)
                 {
-                    if (message?.Data is not null)
-                    {
-                        await _publisher.PublishMessageAsync(message, token)
-                            .ConfigureAwait(false);
-                    }
+                    await _publisher.PublishMessageAsync(message, token)
+                        .ConfigureAwait(false);
```

Comment:
nit: This could probably move into the line above and then we could even remove the parantheses of the `if`, if we want to.